Sources/MapboxCoreNavigation/NativeHandlersFactory.swift

```diff
@@ -43,19 +40,14 @@ class NativeHandlersFactory {
     
     // MARK: - Native Handlers
     
-    lazy var historyRecorder: HistoryRecorderHandle? = {
-        historyDirectoryURL.flatMap {
-            historyRecorderHandlerFactory.getHandler(with: (path: $0.path,
-                                                            configHandle: configHandle),
-                                                     cacheData: self)
-        }
-    }()
-    
     lazy var navigator: MapboxNavigationNative.Navigator = {
         onMainQueueSync { // Make sure that Navigator pick ups Main Thread RunLoop.
             let loggingLevel = NSNumber(value: LoggingLevel.info.rawValue)
             LogConfiguration.setLoggingLevelForUpTo(loggingLevel)
             
+            let historyRecorder = HistoryRecorder.shared.handle
+            let configHandle = Self.configHandle(by: configFactoryType)
```

Comment:
You have different instances of config handle here and in `CacheHandle`. Is it okay?

Reply:
AFAIK, config handle is fine to be allocated multiple times. `CacheHandle` is still cached for re-using which is important